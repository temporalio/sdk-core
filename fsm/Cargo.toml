[package]
name = "rustfsm"
version = "0.1.0"
authors = ["Spencer Judge <sjudge@hey.com>"]
edition = "2018"
license-file = "LICENSE.txt"

[dependencies]
thiserror = "1.0"
derive_more = "0.99"
state_machine_procmacro = { path = "state_machine_procmacro" }
state_machine_trait = { path = "state_machine_trait" }
