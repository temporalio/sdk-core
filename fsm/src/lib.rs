pub use state_machine_procmacro::fsm;
pub use state_machine_trait::{MachineError, StateMachine, TransitionResult};
