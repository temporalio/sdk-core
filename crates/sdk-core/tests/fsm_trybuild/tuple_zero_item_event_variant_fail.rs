use temporalio_macros::fsm;

fsm! {
    name Simple; command SimpleCmd; error Infallible;

    One --(A(), on_a)--> Two
}

fn main() {}
