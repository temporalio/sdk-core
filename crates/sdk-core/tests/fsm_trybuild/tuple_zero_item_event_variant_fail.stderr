error: Only tuple variants with exactly one item are supported for events
 --> tests/fsm_trybuild/tuple_zero_item_event_variant_fail.rs:6:12
  |
6 |     One --(A(), on_a)--> Two
  |            ^
